protos/format.proto

```diff
@@ -134,6 +134,17 @@ message IndexMetadata {
 
   // The version of the dataset this index was built from.
   uint64 dataset_version = 4;
+
+  /// A bitmap of the included fragment ids.
+  ///
+  /// This may by used to determine how much of the dataset is covered by the
+  /// index. This information can be retrieved from the dataset by looking at
+  /// the dataset at `dataset_version`. However, since the old version may be
+  /// deleted while the index is still in use, this information is also stored
+  /// in the index.
+  /// 
+  /// The bitmap is stored as a 32-bit Roaring bitmap.
+  bytes fragment_bitmap = 5;
```

Comment:
What's the implication of this to the latency of opening index if we have 100K+ fragments?

Reply:
In the worst case, there's an additional 1MB of data to download. That's in the absolute worst case, where you don't have any runs of fragment_ids (1, 2, 3, 4), but instead have a bunch of holes (1, 3, 5, 9, ...). So most of the time it will be way more compressed than that, since it just needs to store the range or a set of ranges of ids.

At the very least, we'll want to store the `max_fragment_id` here, because as soon as folks start cleaning up old versions, `knn_combined` will be broken:

https://github.com/lancedb/lance/blob/f3f8794d92f98b5c6f62e5cc2c97e7be0750874f/rust/lance/src/dataset/scanner.rs#L645-L647

Currently, we are always opening the previous version of the dataset, so by having this field we can avoid that. (Previously, `index.dataset_version` referred to the version where the index was added, but after the concurrent writer fixes, it now refers to the version where the index was created from, which is usually--but not always--one version before.) If we add this, we never have to open the old version of the dataset again, so we are actually saving time by not reading that manifest. That'll be a win in almost all cases.

The main reason we use a bitmap instead of just the max id is so we can know how many removed fragments are there in the index. This can happen both from deletion and compaction.

Reply:
Ok, `1MB` sounds fine.